includes/admin/class-sensei-onboarding.php

```diff
@@ -55,34 +83,56 @@ public function __construct() {
 
 			// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Arguments used for comparison.
 			if ( isset( $_GET['page'] ) && ( $_GET['page'] === $this->page_slug ) ) {
-
-				add_action(
-					'admin_print_scripts',
-					function() {
-						Sensei()->assets->enqueue( 'sensei-onboarding', 'onboarding/index.js', [], true );
-					}
-				);
-
-				add_action(
-					'admin_print_styles',
-					function() {
-						Sensei()->assets->enqueue( 'sensei-onboarding', 'onboarding/style.css', [ 'wp-components' ] );
-					}
-				);
-
-				add_filter(
-					'admin_body_class',
-					function( $classes ) {
-						$classes .= ' sensei-wp-admin-fullscreen ';
-						return $classes;
-					}
-				);
-				add_filter( 'show_admin_bar', '__return_false' );
+				$this->setup_onboarding_page();
 			}
 		}
+	}
+
+	/**
+	 * Register an Onboarding admin page.
+	 */
+	public function register_onboarding_page() {
+		if ( current_user_can( 'manage_sensei' ) ) {
+			add_submenu_page(
+				null,
+				__( 'Onboarding', 'sensei-lms' ),
+				__( 'Onboarding', 'sensei-lms' ),
+				'manage_sensei',
+				$this->page_slug,
+				[ $this, 'render_onboarding_page' ]
+			);
+		}
+	}
+
+	/**
+	 * Set up hooks for loading Onboarding page assets.
+	 */
+	public function setup_onboarding_page() {
+		add_action(
+			'admin_print_scripts',
+			function() {
```

Comment:
Generally, we should avoid inline functions for hooks unless we have a good reason. Mainly because it does not allow for the hook to be removed later, and so it reduces customizability.

Reply:
Good point, 500337bd4200b4642684886d5cb77e20255ecd69.